### PR TITLE
feat: Relates to #8221. ethdb Parity Ethereum tool for interacting with the DB on-demand 

### PR DESCRIPTION
**Pull request by ltfschoen**
- [x] You added a brief description of the PR, e.g.:
  - What does it do? Parity Ethereum interaction with the DB on-demand
  - What important points reviewers should know? See notes: https://hackmd.io/cXsUUCcDTQOm0ZF2KWQ6gg?view
- [ ]
  - Is there something left for follow-up PRs?
- [x] You labeled the PR with appropriate labels if you have permissions to do so.
- [x] You mentioned a related issue if this PR related to it, e.g. `Fixes #228` or `Related #1337`.
- [ ] You asked any particular reviewers to review. If you aren't sure, start with GH suggestions.
- [ ] Your PR adheres [the style guide](https://wiki.parity.io/Coding-guide)
  - In particular, mind the maximal line length.
  - There is no commented code checked in unless necessary.
  - Any panickers have a proof or removed.
- [ ] You updated any rustdocs which may have changed

### CODE REVIEW

**Comment by ltfschoen** [comment]
@tomusdrw 

Question 1: i noticed an old comment that Rob made about changing from docopt to clap (since it allows you to have help menus for subcommands too) https://github.com/paritytech/parity-ethereum/issues/2913#issuecomment-300521097. so far the code in this PR just uses docopt. Do you think it would be better to use clap (i've used clap before) https://github.com/paritytech/parity-ethereum/pull/10664/files#diff-6f1ae9a92346a60331f8dff8af38f2cdR30
since it allows help menus for subcommands?

Question 2: should we remove `parity export blocks [OPTIONS] [--] [FILE]` from the Parity CLI and replace it with `ethdb export blocks [OPTIONS] [--] [FILE]` in the ethdb tool?

File: ethdb/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+//! Parity Ethereum command-line tool for database interaction.
+
+#![warn(missing_docs)]
+
+extern crate common_types;
+extern crate serde;
+extern crate docopt;
+
+use common_types::{BlockNumber};
```
Comment (dvdplm) [inline ethdb/src/main.rs]:
```suggestion
use common_types::BlockNumber;
```

File: ethdb/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    --std-err-only     With --std-json redirect to err output only.
+    --std-out-only     With --std-json redirect to out output only.
+    --std-dump-json    Display results in standardized JSON format
+                       with additional state dump.
+    -h, --help         Display this message and exit.
+";
+
+#[derive(Debug, Deserialize)]
+struct Args {
+  flag_chain: String,
```
Comment (dvdplm) [inline ethdb/src/main.rs]: Nitpick: in my head a flag is a binary option (either it's "up" or "down", no mid-mast!). I think you're using it here to distinguish between subcommands and options. I'd suggest s/flag/opt/ and to reverse the order, so s/flag_from/from_opt/

File: ethdb/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+## ethdb
+
+Parity Ethereum interaction with the DB on-demand.
```
Comment (dvdplm) [inline ethdb/README.md]: Probably a dumb question: I'm not sure I understand what "on-demand" means in this context. Can you elaborate?

File: ethdb/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+### Usage
+
+```
+Parity Ethereum interaction with the DB on-demand.
```
Comment (tomusdrw) [inline ethdb/README.md]: That will probably get quickly outdated. If we stick with docopt perhaps we could import that content from the same file used in the binary?

File: ethdb/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Parity Ethereum command-line tool for database interaction.
+
+#![warn(missing_docs)]
+
+extern crate common_types;
```
Comment (tomusdrw) [inline ethdb/src/main.rs]: We should use `edition=2018` in Cargo.toml instead.

File: ethdb/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+extern crate serde;
+extern crate docopt;
+
+use common_types::{BlockNumber};
+
+use docopt::Docopt;
+use serde::Deserialize;
+
+const USAGE: &'static str = "
+Parity Ethereum interaction with the DB on-demand.
```
Comment (tomusdrw) [inline ethdb/src/main.rs]: I'd move this to a separate file and `include_str!` here if we are going to stick with docopt

File: ethdb/src/main.rs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+";
+
+#[derive(Debug, Deserialize)]
+struct Args {
+  flag_chain: String,
+  flag_from: BlockNumber,
+  flag_to: BlockNumber,
+  flag_diff: bool,
+  flag_block: bool,
+  flag_receipts: bool,
```
Comment (tomusdrw) [inline ethdb/src/main.rs]: Seems that this file (and probably other files) use spaces for indentation, please configure your editor to adhere to `.editorconfig` or switch to tabs as per rest of the codebase.

File: ethdb/src/main.rs (hunk 2 above)
Comment (ltfschoen) [reply ethdb/src/main.rs]: i've addressed this in commit aacc0a8b3dd50a0ce9a707a8e88a1741f06de9da

File: ethdb/src/main.rs (hunk 7 above)
Comment (ltfschoen) [reply ethdb/src/main.rs]: i've addressed this in commit b8e0093a0f83334de7224529362c5ef663329d1c

File: ethdb/README.md (hunk 3 above)
Comment (ltfschoen) [reply ethdb/README.md]: i've explained it in the "Experience of New Feature" section of the notes here: https://hackmd.io/cXsUUCcDTQOm0ZF2KWQ6gg?view

File: ethdb/src/main.rs (hunk 6 above)
Comment (ltfschoen) [reply ethdb/src/main.rs]: i've addressed this in commit ee203a618471d34514140fa6188ddbf147d3635c

File: ethdb/src/main.rs (hunk 5 above)
Comment (ltfschoen) [reply ethdb/src/main.rs]: i've addressed this in commit 99b868d25a157528840fd2f46e6e61d638131669

File: ethdb/README.md (hunk 4 above)
Comment (ltfschoen) [reply ethdb/README.md]: i've addressed this in commit ee203a618471d34514140fa6188ddbf147d3635c

**Comment by ltfschoen** [comment]
oh this could be a handy resource https://github.com/paritytech/parity-bridge/blob/master/cli/src/main.rs

**Comment by debris** [comment]
@twittner please reopen when it's ready for another review

**Pull request closed without merging**